### PR TITLE
PR: Launch first console instance only if server is up

### LINKED ISSUE #59: Don't start to create consoles until the server is up and running
@andfoy, you can use this code to keep pinging the server until it's up (at least for 10 secs) and if not, report an error:

https://github.com/spyder-ide/spyder-notebook/blob/master/spyder_notebook/utils/nbopen.py#L65-L70

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #59

### CODE REVIEW

File: spyder_terminal/terminalplugin.py [hunk 1]
```diff
@@ -119,6 +122,35 @@ def __init__(self, parent):
...
+                                   "files.".format(self.port,
+                                                   self.stdout_file,
+                                                   self.stderr_file)),
+                                 QMessageBox.Ok)
+        elif code != 200:
+            self.server_retries += 1
+            QTimer.singleShot(250, self.__wait_server_to_start)
+        elif code == 200:
+            self.create_new_term(give_focus=False)
+            self.sig_server_is_up.emit()
```
Comment (ccordoba12) [inline spyder_terminal/terminalplugin.py]: This signal seems unnecessary, I mean, nothing is connected to it.

Comment (andfoy) [inline spyder_terminal/terminalplugin.py]: I will remove it, then!

File: spyder_terminal/terminalplugin.py [hunk 2]
```diff
@@ -119,6 +122,35 @@ def __init__(self, parent):
...
         new_term_shortcut.setContext(Qt.WidgetWithChildrenShortcut)
 
+        self.__wait_server_to_start()
+
+    # ------ Private API ------------------------------------------
+    def __wait_server_to_start(self):
+        try:
+            code = requests.get('http://127.0.0.1:{0}'.format(
+                self.port)).status_code
+        except Exception:
```
Comment (ccordoba12) [inline spyder_terminal/terminalplugin.py]: Remove `Exception` here because it's not needed.

Comment (andfoy) [inline spyder_terminal/terminalplugin.py]: It is needed, when the server is unavailable it actually launches an Exception

Comment (andfoy) [inline spyder_terminal/terminalplugin.py]:
```
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8000): 
Max retries exceeded with url: / 
(Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7fd0b06a3390>:
 Failed to establish a new connection: [Errno 111] Connection refused',))
```

Comment (ccordoba12) [inline spyder_terminal/terminalplugin.py]: Yeah, but if you only use

```python
try:
    ...
except:
    ...
```

then all exceptions will be captured.

**Comment by ccordoba12** [comment]
Looks good now, thanks @andfoy!

File: spyder_terminal/tests/test_terminal.py [hunk 3]
```diff
@@ -82,8 +85,9 @@ def test_new_terminal(qtbot):
     """Test if a new terminal is added."""
     # Setup widget
     terminal = setup_terminal(qtbot)
-    term = terminal.get_current_term()
+    # terminal.close_term()
```
Comment (ccordoba12) [inline spyder_terminal/tests/test_terminal.py]: Remove comment

File: spyder_terminal/tests/test_terminal.py [hunk 4]
```diff
@@ -70,6 +72,7 @@ def test_terminal_font(qtbot):
 def test_terminal_tab_title(qtbot):
     """Test if terminal tab titles are numbered sequentially."""
     terminal = setup_terminal(qtbot)
+    # terminal.close_term()
```
Comment (ccordoba12) [inline spyder_terminal/tests/test_terminal.py]: Remove comment

File: spyder_terminal/tests/test_terminal.py [hunk 5]
```diff
@@ -56,8 +56,10 @@ def setup_terminal(qtbot):
 def test_terminal_font(qtbot):
     """Test if terminal loads a custom font."""
     terminal = setup_terminal(qtbot)
-    term = terminal.get_current_term()
+    # terminal.close_term()
```
Comment (ccordoba12) [inline spyder_terminal/tests/test_terminal.py]: Remove comment

File: spyder_terminal/terminalplugin.py [hunk 6]
```diff
@@ -119,6 +121,34 @@ def __init__(self, parent):
...
+
+        if self.server_retries == self.MAX_SERVER_CONTACT_RETRIES:
+            QMessageBox.Critical(self, _('Spyder Terminal Error'),
+                                 _("Terminal server could not be located at "
+                                   '<a href="http://127.0.0.1:{0}">'
+                                   'http://127.0.0.1:{0}</a>,'
+                                   ' please restart Spyder on debugging mode '
+                                   "and open an issue with the contents of "
+                                   "<tt>{1}</tt> and <tt>{2}</tt> "
+                                   "files.".format(self.port,
```
Comment (ccordoba12) [inline spyder_terminal/terminalplugin.py]: Let's improve this a bit

    files at {3}.

where `{3}` is

     https://github.com/spyder-ide/spyder-terminal/issues

**Comment by ccordoba12** [comment]
I left a couple of comments, then I think this one is ready.

**Pull request merged**